### PR TITLE
Add design doc

### PR DESCRIPTION
**Pull request by doug-wade**
Summarize why we decided to use request local storage, and how it works.

### CODE REVIEW

**Comment by doug-wade** [comment]
@gigabo bump

bump

**Comment by doug-wade** [comment]
bamp

**Comment by doug-wade** [comment]
bump

**Comment by doug-wade** [comment]
@gigabo bump

File: docs/design.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# Why request-local-storage?
+
+Request local storage handles a particular problem, that of
+[high arity functions](https://en.wikipedia.org/wiki/Arity) in request/response
```
Comment (gigabo) [inline docs/design.md:4]: It's both arity and also _depth_: How many levels of function calls does a parameter need to be passed through to reach where it's actually used?

**Comment by gigabo** [comment]
Yeah, good. 👍 

Thanks @doug-wade.

**Pull request merged**